src/goose/client.clj

```diff
@@ -157,3 +161,11 @@
   - [Periodic Jobs wiki](https://github.com/nilenso/goose/wiki/Periodic-Jobs)"
   [opts cron-opts execute-fn-sym & args]
   (register-cron-schedule opts cron-opts execute-fn-sym args))
+
+(defn perform-batch
+  ([opts execute-fn-sym callback-fn-sym args-coll]
```

Comment:
@siripr4 Let's accept callback-fn-sym as part of the opts? 
Did you have a separate discussion on why it should not be part of opts?

Reply:
Actually, we can keep the signature like `perform-every`. 
`[opts batch-opts execute-fn-sym args-coll]`

Reply:
> Did you have a separate discussion on why it should not be part of opts?

Yes, I and @sezaljain discussed how this doesn't belong in `opts`, since it's not part of how we configure the batch.

> Actually, we can keep the signature like perform-every.
[opts batch-opts execute-fn-sym args-coll]

This makes sense, updated accordingly.
`batch/new` would parse the opts and create a batch.